src/jinja2_fragments/litestar.py

```diff
@@ -0,0 +1,131 @@
+from typing import Any, Dict, Iterable, Optional, Tuple, Union, cast
+
+try:
+    import itertools
+    from mimetypes import guess_type
+    from pathlib import PurePath
+
+    from litestar import Litestar
+    from litestar.background_tasks import BackgroundTask, BackgroundTasks
+    from litestar.connection import Request
+    from litestar.contrib.htmx.response import HTMXTemplate
+    from litestar.contrib.htmx.types import EventAfterType, PushUrlType, ReSwapMethod
+    from litestar.datastructures import Cookie
+    from litestar.enums import MediaType
+    from litestar.exceptions import ImproperlyConfiguredException, LitestarException
+    from litestar.response.base import ASGIResponse
+    from litestar.utils.deprecation import warn_deprecation
+except ModuleNotFoundError as e:
+    raise ModuleNotFoundError(
+        "Install litestar[jinja] before using jinja_fragments.litestar"
+    ) from e
+
+from . import render_block
+
+
+class BlockNotFoundError(LitestarException):
+    def __init__(
+        self, block_name: str, template_name: str, message: Optional[str] = None
+    ):
+        self.block_name = block_name
+        self.template_name = template_name
+        super().__init__(
+            message
+            or f"Block {self.block_name!r} not found in template {self.template_name!r}"
+        )
+
+
+class LitestarHTMXTemplate(HTMXTemplate):
```

Comment:
What do you think of naming this class `HTMXBlockTemplate` or `HTMXFragmentTemplate`? It feels it makes it more specific than the current name.

I see how you are looking to provide a drop-in replacement for `HTMXTemplate`, but having `litestar.LitestarHTMXTemplate` seems a bit repetitive.

Another alternative, if we want to emphasize the "drop-in replacement" intent, is to name it directly `HTMXTemplate`. We would need to change the import above to something like `from litestar.contrib.htmx.response import HTMXTemplate as LitestarHTMXTemplate` or similar to avoid clashing.

Thoughts?

Reply:
Yeah, I think this is the part I struggled with more. I asked a friend if they thought naming it `HTMXTemplate` was good practice or not (for drop in replacement feature), but they recommended naming it something different, and then leaving it up to the user if they want to use the same alias... In that sense, I am happy to change the name to one of the suggestions above, especially since it makes it clear that it is part of this library.

I'm leaning toward `HTMXBlockTemplate` to somewhat mirror the "Block" in the FastAPI implementation. If I don't get to the change tonight, I'll update tomorrow.